presto-iceberg/src/main/java/io/prestosql/iceberg/IcebergSplitManager.java

```diff
@@ -0,0 +1,69 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.iceberg;
+
+import io.prestosql.plugin.hive.HdfsEnvironment;
+import io.prestosql.plugin.hive.HdfsEnvironment.HdfsContext;
+import io.prestosql.plugin.hive.metastore.HiveMetastore;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorSplitSource;
+import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.classloader.ClassLoaderSafeConnectorSplitSource;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+
+import javax.inject.Inject;
+
+import static io.prestosql.iceberg.IcebergUtil.getIcebergTable;
+import static io.prestosql.iceberg.IcebergUtil.getTableScan;
+import static java.util.Objects.requireNonNull;
+
+public class IcebergSplitManager
+        implements ConnectorSplitManager
+{
+    private final IcebergTransactionManager transactionManager;
+    private final HdfsEnvironment hdfsEnvironment;
+
+    @Inject
+    public IcebergSplitManager(IcebergTransactionManager transactionManager, HdfsEnvironment hdfsEnvironment)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableHandle handle, SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) handle;
+
+        HiveMetastore metastore = transactionManager.get(transaction).getMetastore();
+        Configuration configuration = hdfsEnvironment.getConfiguration(new HdfsContext(session, table.getSchemaName()), new Path("file:///tmp"));
+        Table icebergTable = getIcebergTable(table.getSchemaName(), table.getTableName(), configuration, metastore);
+
+        TableScan tableScan = getTableScan(session, table.getPredicate(), table.getSnapshotId(), icebergTable);
+
+        // TODO Use residual. Right now there is no way to propagate residual to presto but at least we can
+        //      propagate it at split level so the parquet pushdown can leverage it.
+        IcebergSplitSource splitSource = new IcebergSplitSource(
+                tableScan.planTasks().iterator(),
```

Comment:
resource leak: `tableScan.planTasks()` returns `CloseableIterator` that needs to be closed.

Reply:
Thanks, fixed.